openstack_hypervisor/utils.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2022 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from netifaces import AF_INET, gateways, ifaddresses
+
+
+def get_local_ip_by_default_route() -> str:
```

Comment:
please push this directly into the hooks module for the moment - we need to split that up but we can do that under a separate PR